lib/hypervisor/hv_kvm/__init__.py

```diff
@@ -2713,6 +2713,89 @@ def ValidateParameters(cls, hvparams):
     kvm_output = cls._GetKVMOutput(kvm_path, cls._KVMOPT_MLIST)
     validate_machine_version(hvparams, kvm_output)
 
+  @classmethod
+  def AssessParameters(cls, hvparams):
+    """Check the given parameters for uncommon/suboptimal values
+
+    This should check the passed set of parameters for suboptimal
+    values.
+
+    @type hvparams: dict
+    @param hvparams: dictionary with parameter names/value
+
+    """
+
+    warnings = []
+
+    cpu_type = hvparams[constants.HV_CPU_TYPE]
+    if not cpu_type:
+      warnings.append("cpu_type is currently unset and defaults to 'qemu64'"
+                      ", please read the gnt-instance man page on the security"
+                      " and performance implications of this parameter.")
+    elif cpu_type in ("qemu32", "qemu64", "kvm32", "kvm64"):
+      warnings.append("cpu_type is currently set to '%s'"
+                      ", please read the gnt-instance man page on the security"
+                      " and performance implications of this parameter." %
+                      cpu_type)
+    elif cpu_type == "host":
+      warnings.append("cpu_type is currently set to 'host', please make"
+                      " sure all your cluster nodes have the exact same"
+                      " CPU type to allow live migrations.")
+
+    disk_aio = hvparams[constants.HV_KVM_DISK_AIO]
+    if disk_aio == "threads":
+      warnings.append("disk_aio is set to 'threads' and should be set"
+                      " to 'native' for better disk performance.")
+
+    vhost_net = hvparams[constants.HV_VHOST_NET]
+    if not vhost_net:
+      warnings.append("vhost_net should be enabled for paravirtual NICs to"
+                      " improve network latency and throughput.")
+
+    migration_bandwidth = hvparams[constants.HV_MIGRATION_BANDWIDTH]
+    if migration_bandwidth < 125:
+      warnings.append("migration_bandwidth is set to less than 125 MiB/s"
+                      ", please consider adjusting it to a higher value"
+                      " that can leverage todays network speeds.")
```

Comment:
Would it be possible to compare against cluster-init-defaults here (`constants.HVC_DEFAULTS.kvm.migration_bandwidth=32`)? And if the cluster has still defaults, remember the admin to tune `migration_bandwidth` and `migration_downtime`? I find it difficult to generally recommended >1Gb/s.

Reply:
sure, that sounds reasonable. Ill add this!